lib/dom.js

```diff
@@ -1175,12 +1189,23 @@ function serializeToString(node,buf,isHTML,nodeFilter,visibleNamespaces){
 		var prefixedNodeName = nodeName
 		if (!isHTML && !node.prefix && node.namespaceURI) {
 			var defaultNS
+			// lookup current default ns from `xmlns` attribute
 			for (var ai = 0; ai < attrs.length; ai++) {
 				if (attrs.item(ai).name === 'xmlns') {
 					defaultNS = attrs.item(ai).value
 					break
 				}
 			}
+			if (!defaultNS) {
+				// lookup current default ns in visibleNamespaces
+				for (var nsi = visibleNamespaces.length - 1; nsi >= 0; nsi--) {
```

Comment:
(minor) nit / question: could we replace some of this code with this kind of pattern:

    Array.from(visibleNamespaces).reverse().find(...)

Reply:
Of course it would be less code, but with additional memory cost. And since this can happen quite often, I would not consider it.